src/BuildPrediction/Predictors/TsConfigPredictor.cs

```diff
@@ -0,0 +1,257 @@
+                        if (!visitedDirectories.Add(directoryToEnumerate))
+                        {
+                            continue;
+                        }
+
+                        // Some globs might point to non-existent paths.
+                        if (!Directory.Exists(directoryToEnumerate))
+                        {
+                            continue;
+                        }
+
+                        foreach (string file in Directory.EnumerateFiles(directoryToEnumerate, "*", SearchOption.TopDirectoryOnly))
+                        {
+                            if (finalGlob.IsMatch(file))
+                            {
+                                predictionReporter.ReportInputFile(file);
+                            }
+                        }
+
+                        foreach (string directory in Directory.EnumerateDirectories(directoryToEnumerate, "*", SearchOption.TopDirectoryOnly))
+                        {
+                            directoriesToEnumerate.Enqueue(directory);
+                        }
+                    }
+                }
+
+                if (isTsConfig && tsConfig.CompilerOptions != null)
+                {
+                    if (!string.IsNullOrEmpty(tsConfig.CompilerOptions.OutFile))
+                    {
+                        string outFileFullPath = PathUtilities.NormalizePath(Path.Combine(configFileDir, tsConfig.CompilerOptions.OutFile));
+                        predictionReporter.ReportOutputFile(outFileFullPath);
+                    }
+
+                    if (!string.IsNullOrEmpty(tsConfig.CompilerOptions.OutDir))
+                    {
+                        string outDirFullPath = PathUtilities.NormalizePath(Path.Combine(configFileDir, tsConfig.CompilerOptions.OutDir));
+                        predictionReporter.ReportOutputDirectory(outDirFullPath);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// This mostly follows the logic of the FindConfigFiles target and task.
+        /// There is some probing behavior when there are no Content or TypeScriptCompile items, but we're not currently supporting that scenario.
+        /// </summary>
+        private List<string> FindConfigFiles(ProjectInstance projectInstance)
+        {
+            ICollection<ProjectItemInstance> contentItems = projectInstance.GetItems(ContentItemName);
+            ICollection<ProjectItemInstance> typeScriptCompileItems = projectInstance.GetItems(TypeScriptCompileItemName);
+
+            static bool IsConfigFile(string file)
+            {
+                if (file == null)
+                {
+                    return false;
+                }
+
+                string fileName = Path.GetFileName(file);
+                return fileName.Equals(TsConfigFileName, PathComparer.Comparison)
+                    || fileName.Equals(JsConfigFileName, PathComparer.Comparison);
```

Comment:
Is this same as tsconfig?

Reply:
Almost. It's a subset.

See: https://code.visualstudio.com/docs/languages/jsconfig

The build targets do look for and handle them as well, which is why they're here.